ql/instruments/basketoption.hpp

```diff
@@ -36,7 +36,7 @@ namespace QuantLib {
       private:
         boost::shared_ptr<Payoff> basePayoff_;
       public:
-        BasketPayoff(const boost::shared_ptr<Payoff> &p)
+        explicit BasketPayoff(const boost::shared_ptr<Payoff> &p)
```

Comment:
Yes, all additions of `explicit` are breaks, but let's do it.  I'd bet a few bucks that, even in client code, all calls to the constructors of these classes are of the form `shared_ptr<Foo>(new Foo(arg))` and there are no implicit conversions.